addons/barfiller/statics.lua

```diff
@@ -0,0 +1,844 @@
+end
+
+function get_rested_bonus_pos_y()
+    return bonus_pos_y
+end
+
+function get_rested_bonus_visible()
+    return bonus_visible
+end
+
+-- Background Bar Setters
+function set_bg_pos_x(new_pos_x)
+    bg_pos_x = new_pos_x
+    background_bar:pos(new_pos_x, get_bg_pos_y())
+end
+
+function set_bg_pos_y(new_pos_y)
+    bg_pos_y = new_pos_y
+    background_bar:pos(get_bg_pos_x(), new_pos_y)
+end
+
+function set_bg_visible(new_visible_state)
+    bg_visible = new_visible_state
+    background_bar:visible(new_visible_state)
+end
+
+function set_bg_alpha(new_alpha)
+    bg_alpha = new_alpha
+    background_bar:alpha(new_alpha)
+end
+
+function set_bg_red(new_red)
+    bg_red = new_red
+    background_bar:color(new_red, get_bg_green(), get_bg_blue())
+end
+
+function set_bg_green(new_green)
+    bg_green = new_green
+    background_bar:color(get_bg_red(), new_green, get_bg_blue())
+end
+
+function set_bg_blue(new_blue)
+    bg_blue = new_blue
+    background_bar:color(get_bg_red(), get_bg_green(), new_blue)
+end
+
+function set_bg_image(new_path)
+    bg_image = new_path
+    background_bar:path(new_path)
+end
+
+function set_bg_fit(new_fit)
+    bg_fit = new_fit
+    background_bar:fit(new_fit)
+end
+
+function set_bg_width(new_width)
+    bg_width = new_width
+    background_bar:size(new_width, get_bg_height())
+end
+
+function set_bg_height(new_height)
+    bg_height = new_height
+    background_bar:size(get_bg_width(), new_height)
+end
+
+function set_bg_repeat_x(new_repeat_x)
+    bg_repeat_x = new_repeat_x
+    background_bar:repeat_xy(new_repeat_x, get_bg_repeat_y())
+end
+
+function set_bg_repeat_y(new_repeat_y)
+    bg_repeat_y = new_repeat_y
+    background_bar:repeat_xy(get_bg_repeat_x(), new_repeat_y)
+end
+
+-- Foreground Bar Setters
+function set_fg_pos_x(new_pos_x)
+    fg_pos_x = new_pos_x
+    foreground_bar:pos(new_pos_x, get_fg_pos_y())
+end
+
+function set_fg_pos_y(new_pos_y)
+    fg_pos_y = new_pos_y
+    foreground_bar:pos(get_fg_pos_x(), new_pos_y)
+end
+
+function set_fg_visible(new_visible_state)
+    fg_visible = new_visible_state
+    foreground_bar:visible(new_visible_state)
+end
+
+function set_fg_alpha(new_alpha)
+    fg_alpha = new_alpha
+    foreground_bar:alpha(new_alpha)
+end
+
+function set_fg_red(new_red)
+    fg_red = new_red
+    foreground_bar:color(new_red, get_fg_green(), get_fg_blue())
+end
+
+function set_fg_green(new_green)
+    fg_green = new_green
+    foreground_bar:color(get_fg_red(), new_green, get_fg_blue())
+end
+
+function set_fg_blue(new_blue)
+    fg_blue = new_blue
+    foreground_bar:color(get_fg_red(), get_fg_green(), new_blue)
+end
+
+function set_fg_image(new_path)
+    fg_image = new_path
+    foreground_bar:path(new_path)
+end
+
+function set_fg_fit(new_fit)
+    fg_fit = new_fit
+    foreground_bar:fit(new_fit)
+end
+
+function set_fg_width(new_width)
+    fg_width = new_width
+    foreground_bar:size(new_width, get_fg_height())
+    if frame_count%15 == 0 then update_strings() end
```

Comment:
I believe this should be done with a timer instead of a frame counter. If someone is playing with uncapped frame rate, this may still update seven times a second. If someone is lagging it may only update once a second or only every few seconds. If you did this with a timer (i.e. update if half a second has passed since the last time you did it) it would be more consistent:
```lua
do
    local last_update = 0

    function set_fg_width(new_width)
        fg_width = new_width
        foreground_bar:size(new_width, get_fg_height())

        local now = os.clock()
        if now - last_update > 0.5 then
            update_strings()
            last_update = now
        end
    end
end
```